File: tests/fixtures/edgecase_repo/rust/src/module4.rs
use crate::module3::Service3;

pub struct Base4 {
    pub name: String,
}

pub struct Service4 {
    pub base: Base4,
}

pub trait Handler4 {
    fn handle(&self, payload: &str) -> bool;
}

impl Handler4 for Service4 {
    fn handle(&self, payload: &str) -> bool {
        self.validate(payload);
        true
    }
}

impl Service4 {
    fn validate(&self, payload: &str) {}
}

pub fn helper_4(value: &str) -> &str {
    value
}

File: tests/fixtures/edgecase_repo/typescript/src/module4.ts
import { Service3 } from "./module3";

export interface Handler4 {
  handle(payload: string): boolean;
}

export class Base4 {
  protected name: string;
  constructor(name: string) {
    this.name = name;
  }
}

export class Service4 extends Base4 implements Handler4 {
  handle(payload: string): boolean {
    this.validate(payload);
    return new Service3("x").handle(payload);
  }

  private validate(payload: string): void {}
}

export function helper4(value: string): string {
  return value;
}
